From 9ff8ccc7cbc48c797d6fc214638e24f2aadb8c5a Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sat, 22 Nov 2014 22:27:10 +0000
Subject: [PATCH 7/7] Search for freeaddrinfo in SOCKET_LIBS

.. using system-specifc headers so that the declaration
includes the correct calling convention.
---
 configure.ac | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/configure.ac b/configure.ac
index d576265..6e4656f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -649,6 +649,16 @@ AC_LINK_IFELSE([
         inet_pton(AF_INET, "127.0.0.1", dst);
     ])],[AC_DEFINE([HAVE_INET_PTON],[1],[Define to 1 if you have inet_pton function])],[AC_LIBOBJ([inet_pton])])
 AC_CHECK_FUNCS([if_nameindex if_nametoindex])
+AC_LINK_IFELSE(
+    [AC_LANG_PROGRAM([#ifdef _WIN32
+                     #include <winsock2.h>
+                     #include <ws2tcpip.h>
+                     #else
+                     #include <sys/socket.h>
+                     #include <netdb.h>
+                     #endif], [
+        freeaddrinfo(0);
+    ])],[AC_DEFINE([HAVE_FREEADDRINFO],[1],[Define to 1 if you have freeaddrinfo function])],[AC_LIBOBJ([freeaddrinfo])])
 VLC_RESTORE_FLAGS
 
 AS_IF([test -n "$SOCKET_LIBS"], [
-- 
2.1.3

